### PR TITLE
refactor(connector): [Shift4] change error message from NotSupported to NotImplemented

### LINKED ISSUE #2860: [REFACTOR] : [Shift4] Error Message For Connector Implementation
### :memo: Feature Description

- In terms of errors, we are currently throwing a 'not supported' message, which shouldn't be the case as we are yet to decide on the implementation.

### :hammer: Possible Implementation

- In order to manage payment methods that are not implemented by Hyperswitch or yet to be implemented, the connector transformers file should make use of the NotImplemented ConnectorError enum variant.
- By doing so, we will throw same error message for all the Connector Implementation 
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2831

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by unpervertedkid**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Resolves #2860 

## How did you test it?

No test cases required. As In this PR only error message have been changed from Not Supported error message to NotImplemented error message.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by swangi-kumari** [comment]
Hey @unpervertedkid ,
Please resolve the conflicts from your branch.

File: crates/router/src/connector/shift4/transformers.rs [hunk 1]
```diff
@@ -286,10 +282,9 @@ impl TryFrom<&api_models::payments::GiftCardData> for Shift4PaymentMethod {
     type Error = Error;
     fn try_from(gift_card_data: &api_models::payments::GiftCardData) -> Result<Self, Self::Error> {
         match gift_card_data {
-            payments::GiftCardData::Givex(_) => Err(errors::ConnectorError::NotSupported {
-                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
-                connector: "Shift4",
-            }
+            payments::GiftCardData::Givex(_) => Err(errors::ConnectorError::NotImplemented(
```
Comment (swangi-kumari) [inline crates/router/src/connector/shift4/transformers.rs:285]: Here, as `givex` and `paysafecard` are throwing the same error now.
Could you please OR ( | ) between them and avoid the repetition of same error.

File: crates/router/src/connector/shift4/transformers.rs [hunk 2]
```diff
@@ -273,10 +270,9 @@ impl TryFrom<&api_models::payments::VoucherData> for Shift4PaymentMethod {
             | payments::VoucherData::MiniStop(_)
             | payments::VoucherData::FamilyMart(_)
             | payments::VoucherData::Seicomart(_)
-            | payments::VoucherData::PayEasy(_) => Err(errors::ConnectorError::NotSupported {
-                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
-                connector: "Shift4",
-            }
+            | payments::VoucherData::PayEasy(_) => Err(errors::ConnectorError::NotImplemented(
```
Comment (swangi-kumari) [inline crates/router/src/connector/shift4/transformers.rs:273]: Here, use OR( | ) between `Boleto` and `Efecty` Payment methods as we have common error message now for all the Payment Methods.

File: crates/router/src/connector/shift4/transformers.rs [hunk 3]
```diff
@@ -244,10 +242,9 @@ impl TryFrom<&api_models::payments::BankTransferData> for Shift4PaymentMethod {
             | payments::BankTransferData::DanamonVaBankTransfer { .. }
             | payments::BankTransferData::MandiriVaBankTransfer { .. }
             | payments::BankTransferData::Pix {}
-            | payments::BankTransferData::Pse {} => Err(errors::ConnectorError::NotSupported {
-                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
-                connector: "Shift4",
-            }
+            | payments::BankTransferData::Pse {} => Err(errors::ConnectorError::NotImplemented(
```
Comment (swangi-kumari) [inline crates/router/src/connector/shift4/transformers.rs:245]: Here, use OR( | ) between `MultibancoBankTransfer` and `AchBankTransfer` Payment methods as we have common error message now for all the Payment Methods.

File: crates/router/src/connector/shift4/transformers.rs [hunk 4]
```diff
@@ -212,10 +211,9 @@ impl TryFrom<&api_models::payments::WalletData> for Shift4PaymentMethod {
             | payments::WalletData::TouchNGoRedirect(_)
             | payments::WalletData::WeChatPayQr(_)
             | payments::WalletData::CashappQr(_)
-            | payments::WalletData::SwishQr(_) => Err(errors::ConnectorError::NotSupported {
-                message: utils::SELECTED_PAYMENT_METHOD.to_string(),
-                connector: "Shift4",
-            }
+            | payments::WalletData::SwishQr(_) => Err(errors::ConnectorError::NotImplemented(
```
Comment (swangi-kumari) [inline crates/router/src/connector/shift4/transformers.rs:214]: Here, use OR( | ) between `WeChatPayRedirect` and `AliPayQr` Payment methods as we have common error message now for all the Payment Methods.

**Review by swangi-kumari — CHANGES_REQUESTED** [review]

**Comment by unpervertedkid** [comment]
@swangi-kumari take a look at this iteration

**Review by swangi-kumari — APPROVED** [review]
LGTM!

**Pull request merged**